source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTextBlockRenderer.mm

```diff
@@ -195,6 +196,40 @@ + (NSNumber *)getTextStrokeWidthForWeight:(TextWeight)weight
             return @-2;
     }
 }
+// given string with short form of date (MM/dd/yyyy), convert it to
+// NSDateFormatterMediumStyle or NSDateFormatterLongStyle in local language
++ (void) getLocalizedDate:(std::shared_ptr<TextBlock> const &)txtBlck stringWithDate:(std::string &)dateParsedString
+{
+    std::vector<std::shared_ptr<DateTimePreparsedToken>> DateTimePreparsedTokens =  DateTimePreparser(txtBlck->GetText()).GetTextTokens();
+    for(auto section : DateTimePreparsedTokens){
+        if(section->GetFormat() != DateTimePreparsedTokenFormat::RegularString) {
+            NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+            [formatter setDateFormat:@"MM/dd/yyy"];
+            std::string givenDate = std::to_string(section->GetMonth()) + "/" +  std::to_string(section->GetDay()) + "/" +  std::to_string(section->GetYear());
+            NSString *nsString = [NSString stringWithCString:givenDate.c_str() encoding:NSUTF8StringEncoding];
+            NSDate *date = [formatter dateFromString:nsString];
+            // specify output date format
+            NSDateFormatter *outputFormatter = [[NSDateFormatter alloc] init];
+            outputFormatter.timeStyle = NSDateFormatterNoStyle;
+            if(section->GetFormat() == DateTimePreparsedTokenFormat::DateCompact){
+                outputFormatter.dateStyle = NSDateFormatterShortStyle;
+            } else if(section->GetFormat() == DateTimePreparsedTokenFormat::DateShort){
+                outputFormatter.dateStyle = NSDateFormatterMediumStyle;
+            } else{
+                outputFormatter.dateStyle = NSDateFormatterLongStyle;
+            }
+
+            NSString *languageType= [NSString stringWithCString:txtBlck->GetLanguage().c_str() encoding:NSUTF8StringEncoding];
+            if([languageType compare:@""]){
+                outputFormatter.locale = [[NSLocale alloc] initWithLocaleIdentifier:languageType];
+            }
+            NSString *dateInLocalLanguage = [outputFormatter stringFromDate:date];
+            dateParsedString += std::string([dateInLocalLanguage UTF8String]);
```

Comment:
>std::string( [](start = 32, length = 12)

There's no need to create an intermediate std::string here. You can pass the const char* directly into operator+=.

Reply:
awesome; 

---
In reply to: [171687778](https://github.com/Microsoft/AdaptiveCards/pull/1194#discussion_r171687778) [](ancestors = 171687778)